src/che-workspace-project-manager.ts

```diff
@@ -0,0 +1,69 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { TheiaCloneCommand } from './theia-commands';
+import { CheWorkspaceApi, CheWorkspace } from './che-api';
+import * as theia from '@theia/plugin';
+const fs = require('fs');
+
+/**
+ * Make synchronization between projects defined in Che workspace and theia projects.
+ */
+export class CheWorkspaceProjectManager {
+
+    private readonly cheWorkspaceApi: CheWorkspaceApi;
+
+    constructor() {
+        this.cheWorkspaceApi = new CheWorkspaceApi();
+    }
+
+    async onStart() {
+        const workspaceConfig = await this.getCurrentWorkspaceConfigFromCheMaster();
+        if (!workspaceConfig) {
+            theia.window.showErrorMessage('No Che workspace could be found');
+            return;
+        }
+
+        const cloneCommandList = await this.selectProjectToCloneCommands(workspaceConfig);
+        await this.executeCloneCommands(cloneCommandList);
+
+    }
+
+    private async getCurrentWorkspaceConfigFromCheMaster(): Promise<CheWorkspace | undefined> {
+        return await this.cheWorkspaceApi.retrieveWorkspaceDefinition();
+    }
+
+    async selectProjectToCloneCommands(workspaceConfig: CheWorkspace) {

```

Comment:
Up to you, but I like when return type (: Promise<TheiaCloneCommand[]> in this case) is exist. I omit only Promise<void>.

Reply:
@AndrienkoAleksandr OK